src/test/e2e-emulated/receiver-precedence.spec.ts

```diff
@@ -0,0 +1,234 @@
+import { beginCell, Cell, toNano } from "@ton/core";
+import { Blockchain, SandboxContract, TreasuryContract } from "@ton/sandbox";
+import { ReceiverTester } from "./contracts/output/receiver-precedence_ReceiverTester";
+import { Calculator } from "./contracts/output/receiver-precedence_Calculator";
+import "@ton/test-utils";
+
+describe("receivers-precedence", () => {
+    let blockchain: Blockchain;
+    let treasure: SandboxContract<TreasuryContract>;
+    let contract: SandboxContract<ReceiverTester>;
+    let calculator: SandboxContract<Calculator>;
+
+    beforeEach(async () => {
+        blockchain = await Blockchain.create();
+        blockchain.verbosity.print = false;
+        treasure = await blockchain.treasury("treasure");
+
+        contract = blockchain.openContract(await ReceiverTester.fromInit());
+        calculator = blockchain.openContract(await Calculator.fromInit());
+
+        const deployResult = await contract.send(
+            treasure.getSender(),
+            { value: toNano("10") },
+            null,
+        );
+        expect(deployResult.transactions).toHaveTransaction({
+            from: treasure.address,
+            to: contract.address,
+            success: true,
+            deploy: true,
+        });
+
+        const calcDeploy = await calculator.send(
+            treasure.getSender(),
+            { value: toNano("10") },
+            "deploy",
+        );
+        expect(calcDeploy.transactions).toHaveTransaction({
+            from: treasure.address,
+            to: calculator.address,
+            success: true,
+            deploy: true,
+        });
+    });
+
+    it("should implement receivers precedence correctly", async () => {
+        // Initially, since we sent a deploy message with empty message, the empty receiver executed
+        const receiver1 = await contract.getReceiverKind();
+        expect(receiver1 === "empty").toBe(true);
```

Comment:
Why not `expect(receiver1).toBe("empty");`? In case of errors we will see what we actually got